charts/revere/templates/networkPolicy.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ .Release.name }}-networkpolicy
+  labels:
+    {{- include "revere.labels" . | nindent 4 }}
+spec:
+  policyTypes:
+  {{- if .Values.networkPolicy.ingress.enabled }}
+  - Ingress
+  {{- end }}
+  {{- if .Values.networkPolicy.egress.enabled }}
+  - Egress
+  {{- end }}
+  podSelector:
+    matchLabels:
+      {{- include "revere.labels" . | nindent 6 }}
+
+  {{- if .Values.networkPolicy.ingress.enabled }}
+  ingress:
+  - from:
+    {{- range .Values.networkPolicy.ingress.cidrAllowList }}
+    - ipBlock:
+        cidr: {{ . }}
+    {{- end }}
+    ports:
+      {{- range .Values.networkPolicy.ingress.portAllowList }}
+      - {{- range $k, $v := .}}
+        {{ $k }}: {{ $v }}
+        {{- end }}
+      {{- end }}
+  {{- end }}
+
+  {{- if .Values.networkPolicy.egress.enabled }}
+  egress:
+  - to:
+    {{- range .Values.networkPolicy.egress.cidrAllowList }}
+    - ipBlock:
+        cidr: {{ . }}
+    {{- end }}
+    {{- if .Values.secrets.gcpServiceAccount.workloadIdentity.enabled }}
+    # GKE Metadata Server
+    - ipBlock:
+        cidr: 127.0.0.1/32
+    {{- end }}
+    # DNS
+    - namespaceSelector:
+        matchLabels:
+          networking/namespace: kube-system
+      podSelector:
+        matchLabels:
+          k8s-app: kube-dns
+    ports:
+      {{- range .Values.networkPolicy.egress.portAllowList }}
+      - {{- range $k, $v := . }}
+        {{ $k }}: {{ $v }}
+        {{- end }}
+      {{- end }}
+      {{- if .Values.secrets.gcpServiceAccount.workloadIdentity.enabled }}
+      # GKE Metadata Server
+      - port: 988
+        protocol: TCP
+      {{- end }}
+      # DNS
+      - port: 53
+        protocol: TCP
+      - port: 53
+        protocol: UDP
```

Comment:
This is an explicit allow rather than explicit deny correct?

Reply:
Yeah, the second a network policy for ingress or egress applies to a pod, ingress or egress is denied and then you have to add it back in here. It is separate for ingress and egress, at least.